edgelet/dps/src/registration.rs

```diff
@@ -593,13 +591,14 @@ mod tests {
     use std::sync::Mutex;
 
     use edgelet_core::crypto::{MemoryKey, MemoryKeyStore};
-    use http;
     use hyper::{self, Body, Request, Response, StatusCode};
-    use serde_json;
-    use tokio;
     use url::Url;
 
-    use super::*;
+    use super::{
```

Comment:
Although this compiles and works it is confusing now. `super` module (parent) which is `registration` does not have a child `future` it just imports `future` from futures crate.
It'd be really nice if we can avoid this confusion by importing with appropriate crate: not `super::future` but `futures::future`.

Reply:
This comment applies to all files

Reply:
doesn't this the same problem of existing code?  And should the test always use whatever the parent uses?  This recalls 1 thing I read from "Rust programming language" book, it depends on how do you want to organize your tests.  If you want tests stick with implementation in the same file, uses super seems more appropriate.  If we plan to move tests to its own folder, then state references explicitly.

Since this doesn't change existing behavior, can this be handled in another PR? and we should make sure this is what our team is going to follow.

Reply:
I agree with Denis - usage of external crates shouldn't be imported from "super".  I would suggest creating an issue and assign owners as appropriate for each section.

Reply:
created a bug https://dev.azure.com/msazure/One/_workitems/edit/7123592 to track.